### PR TITLE
Feature: batch evaluate exclude not found

### PR DESCRIPTION
**Pull request by itaischwartz**
Add an option to exclude not-found flags from batch evaluate calls.

This change makes it possible to separate flags management from client calls.
- a client can add a flag to the batchEvaluateRequest list even before the flag was added.
- admin can remove flags w/o having to worry about batchEvaluate calls failing altogether.

### CODE REVIEW

**Comment by itaischwartz** [comment]
@markphelps 
We can think of a new version of batchEvaluate (batchEvaluteV2) where we expose a separate error value for each flag in responseList, and let the client choose how to handle the error.

File: server/evaluator_test.go [hunk 1]
```diff
@@ -91,6 +91,55 @@ func TestEvaluate_FlagNotFound(t *testing.T) {
...
+	}
+	store.On("GetFlag", mock.Anything, "foo").Return(enabledFlag, nil)
+	store.On("GetFlag", mock.Anything, "bar").Return(disabled, nil)
+	store.On("GetFlag", mock.Anything, "NotFoundFlag").Return(&flipt.Flag{}, errors.ErrNotFoundf(""))
+
+	store.On("GetEvaluationRules", mock.Anything, "foo").Return([]*storage.EvaluationRule{}, nil)
+
+	resp, err := s.BatchEvaluate(context.TODO(), &flipt.BatchEvaluationRequest{
+		RequestId:       "12345",
+		ExcludeNotFound: true,
```
Comment (markphelps) [inline server/evaluator_test.go:115]: would you mind adding another test where `ExcludeNotFound` is false or simply not set (so it defaults to false)?

**Review by markphelps — COMMENTED** [review]
Great idea @itaischwartz ! Thanks for the contribution. Just one minor request re: tests.

I agree maybe a new API method is needed to add an error to each evaluation request/response.

**Comment by markphelps** [comment]
@all-contributors please add @itaischwartz for code

File: server/evaluator_test.go (hunk 1 above)
Comment (itaischwartz) [reply server/evaluator_test.go:115]: I think that `TestEvaluate_FlagNotFound` is testing this case.

Comment (itaischwartz) [reply server/evaluator_test.go:115]: I see now that it's testing another grpc call.
I'll add the missing test case

Comment (itaischwartz) [reply server/evaluator_test.go]: done

**Review by markphelps — APPROVED** [review]
Awesome! Thanks again!

**Pull request merged**